### PR TITLE
Equippables addition/removal feature

### PR DESCRIPTION
**Pull request by Szegoo**
Closes: #97

### CODE REVIEW

**Comment by Szegoo** [comment]
@bmacer PTAL.

File: traits/src/base.rs [hunk 1]
```diff
@@ -37,6 +37,15 @@ pub struct BaseInfo<AccountId, BoundedString> {
 	pub symbol: BoundedString,
 }
 
+pub enum EquippableType<CollectionId, BoundedCollectionList> {
```
Comment (Szegoo) [inline traits/src/base.rs:40]: Not really proud of the naming here, could use some ideas.

File: traits/src/base.rs [hunk 2]
```diff
@@ -77,7 +86,7 @@ pub trait Base<
 		issuer: AccountId,
 		base_id: BaseId,
 		slot: SlotId,
-		equippables: EquippableList<BoundedCollectionList>,
+		ty: EquippableType<CollectionId, BoundedCollectionList>,
```
Comment (bmacer) [inline traits/src/base.rs:89]: maybe change `ty` to `operation` or `equippable_operation`?

File: pallets/rmrk-equip/src/functions.rs [hunk 3]
```diff
@@ -430,7 +430,10 @@ where
 		issuer: T::AccountId,
 		base_id: BaseId,
 		part_id: PartId,
-		equippables: EquippableList<BoundedVec<CollectionId, T::MaxCollectionsEquippablePerPart>>,
+		ty: EquippableType<
```
Comment (bmacer) [inline pallets/rmrk-equip/src/functions.rs:433]: maybe change `ty` to `operation` or `equippable_operation`?

**Review by bmacer — CHANGES_REQUESTED** [review]
Looks very good, just the `ty` naming is a little meh.

One consideration, this only allows for adding or removing one collection at a time.  Would it be important/necessary to allow a bounded list of collections to add/remove in batches?  @ilionic

**Comment by ilionic** [comment]
@bmacer not important / necessary to add in batches. So all good. 
One improvement @Szegoo if you are comfortable with Typescript and Polkadot js is to add integration test in `/tests` folder.
But it's optional.

**Comment by Szegoo** [comment]
@ilionic Sure I will do that.

File: tests/src/util/tx.ts [hunk 4]
```diff
@@ -810,6 +810,107 @@ export async function setEquippableList(
...
+    slotId,
+    "Error: invalid base ID from update equippable event"
+  );
+
+  const fetchedEquippableList = await getEquippableList(api, baseId, slotId);
+  expect(fetchedEquippableList, "Error: unable to fetch equippable list").to.be
+    .not.null;
+  if (fetchedEquippableList) {
+    // @ts-ignore
+    expect(fetchedEquippableList.Custom).to.be.deep.contain(
```
Comment (Szegoo) [inline tests/src/util/tx.ts:856]: Not the proudest of using `@ts-ignore`. Could use some ideas on how to improve this.

**Comment by Szegoo** [comment]
@ilionic PTAL.

File: tests/src/util/tx.ts (hunk 4 above)
Comment (ilionic) [reply tests/src/util/tx.ts:856]: @Szegoo I think you need to start a chain and generate new metadata with types. In this case `Custom` would be part of generated types. 
https://github.com/rmrk-team/rmrk-substrate/blob/main/tests/package.json#L52
lmk if you have any questions

Comment (Szegoo) [reply tests/src/util/tx.ts]: @ilionic Removed `@ts-ignore`, it should probably be fine now.

**Review by ilionic — APPROVED** [review]
LGTM

File: pallets/rmrk-equip/src/functions.rs [hunk 5]
```diff
@@ -462,8 +465,26 @@ where
...
 					},
 					PartType::SlotPart(mut slot_part) => {
-						// Update equippable value
-						slot_part.equippable = equippables;
+						match operation {
+							EquippableOperation::Add(equippable) => {
+								if let EquippableList::Custom(mut equippables) =
+									slot_part.equippable
+								{
+									let _ = equippables.try_push(equippable);
```
Comment (HashWarlock) [inline pallets/rmrk-equip/src/functions.rs:473]: What happens if this returns `Error` bc the BoundVec exceeds the length?

Comment (Szegoo) [reply pallets/rmrk-equip/src/functions.rs]: Yeah, this should be handled, fixed it now.

**Review by HashWarlock — APPROVED** [review]
LGTM

**Pull request merged**